packages/aws-cdk/lib/diff.ts

```diff
@@ -23,9 +23,10 @@ export function printStackDiff(
   context: number,
   quiet: boolean,
   changeSet?: CloudFormation.DescribeChangeSetOutput,
-  stream?: cfnDiff.FormatStream): number {
+  stream?: cfnDiff.FormatStream,
+  stackDiff?: cfnDiff.TemplateDiff): number {
```

Comment:
Since we need to get the presence or absence of diffs when controlling the output of stack name, we need to call the `cfnDiff.fullDiff` function in the `CdkToolkit.diff` method.
I changed it so that the diff can be passed in optionally so that the relatively heavy processing of `cfnDiff.fullDiff` does not need to be executed multiple times.